lib/sequel/plugins/enum_values.rb

```diff
@@ -68,20 +71,24 @@ def enum_values(field)
 
 				private
 
-				def transform_predicate_methods_to_enum_fields(predicate_methods)
-					case predicate_methods
-					when TrueClass
-						all_enum_fields.keys
-					when FalseClass
-						[]
-					else
-						Array(predicate_methods)
+				def process_enum_values_options
+					process_enum_values_option(:filter_methods) do |field, value|
+						dataset_module { where "#{field}_#{value}", field => value }
+					end
+
+					process_enum_values_option(:predicate_methods) do |field, value|
+						define_method("#{value}?", -> { public_send(field) == value })
 					end
 				end
 
-				def define_predicate_method(field, enum_value)
-					define_method "#{enum_value}?" do
-						public_send(field) == enum_value
+				def process_enum_values_option(key)
+					return unless block_given? && (option = @enum_values_options[key])
+
+					enum_fields = all_enum_fields
+					enum_fields = enum_fields.slice(*Array(option)) unless option == true
```

Comment:
As I see, we don't need for `Array()` here: https://repl.it/repls/UselessElaborateSpof

Reply:
thank you. changed.